### PR TITLE
fixed https://github.com/NuGet/Home/issues/2357

### PR DESCRIPTION
**Pull request by zhili1208**
https://github.com/NuGet/Home/issues/2357

merged all v2, v3 resource to use same getAsync() API from HttpSource

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/V2FeedParser.cs [hunk 1]
```diff
@@ -158,6 +160,7 @@ public V2FeedParser(HttpSource httpSource, PackageSource source)
         /// </summary>
         public async Task<IReadOnlyList<V2FeedPackageInfo>> FindPackagesByIdAsync(
             string id,
+            V2FeedCacheContext cacheContext,
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/V2FeedParser.cs]: How about throwing if this is null and adding a SourceCacheContext.NoCache static property that returns a cache context that contains options to disable caching (same as null does here)

**Comment by emgarten** [comment]
:watch: waiting on null changes

File: src/NuGet.Core/NuGet.DependencyResolver/SourceRepositoryDependencyProvider.cs [hunk 2]
```diff
@@ -25,6 +26,16 @@ public class SourceRepositoryDependencyProvider : IRemoteDependencyProvider
...
 
+        // Limiting concurrent requests to limit the amount of files open at a time on Mac OSX
+        // the default is 256 which is easy to hit if we don't limit concurrency
+        private readonly static SemaphoreSlim _throttle =
+            RuntimeEnvironmentHelper.IsMacOSX
+                ? new SemaphoreSlim(ConcurrencyLimit, ConcurrencyLimit)
+                : null;
+
+        // In order to avoid too many open files error, set concurrent requests number to 16 on Mac
+        private readonly static int ConcurrencyLimit = 16;
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.DependencyResolver/SourceRepositoryDependencyProvider.cs]: private const int

**Comment by yishaigalatzer** [comment]
Please do not merge to 3.5RC or 3.4.2 - 3.5 RTM only

**Comment by emgarten** [comment]
What's the plan for this?

**Comment by zhili1208** [comment]
Need a new PR for this, close it

**Pull request closed without merging**